src/components/touchables/splitStyleProp.ts

```diff
@@ -0,0 +1,61 @@
+import { FlexStyle, StyleProp, StyleSheet } from 'react-native';
+
+const OUTER_PROPS: { [key in keyof FlexStyle]?: true } = {
+  alignSelf: true,
+  bottom: true,
+  display: true,
+  end: true,
+  flex: true,
+  flexBasis: true,
+  flexGrow: true,
+  flexShrink: true,
+  height: true,
+  left: true,
+  margin: true,
+  marginBottom: true,
+  marginEnd: true,
+  marginHorizontal: true,
+  marginLeft: true,
+  marginRight: true,
+  marginStart: true,
+  marginTop: true,
+  marginVertical: true,
+  maxHeight: true,
+  maxWidth: true,
+  minHeight: true,
+  minWidth: true,
+  position: true,
+  right: true,
+  start: true,
+  top: true,
+  width: true,
+  zIndex: true,
+};
+
+/**
+ * Split a style prop between an "outer" and "inner" views in a way that makes it behave
+ * as if it was a single view.
+ *
+ * @example
+ * const { outer, inner } = splitStyleProp(style);
+ * return (
+ *   <View style={outer}>
+ *     <View style={[{ flexGrow: 1 }, inner]} />
+ *   </View>
+ * );
+ */
+export default function splitStyleProp<T extends FlexStyle>(
```

Comment:
I think this should be `splitStyleProp<T extends ViewStyle>` for better clarity.